test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -41,6 +41,50 @@ func invalidDiffWrtFunction(_ fn: @differentiable(Float) -> Float) -> Float {
   return fn(.pi)
 }
 
+// expected-error @+1 {{'invalidDiffNoParams()' has no parameters to differentiate with respect to}}
+@differentiable
+func invalidDiffNoParams() -> Float {
+  return 1
+}
+
+// expected-error @+1 {{cannot differentiate void function 'invalidDiffVoidResult(x:)'}}
```

Comment:
I don’t believe “void function” is a formal concept in Swift. Swift formally calls it “function without return values”. 

See https://docs.swift.org/swift-book/LanguageGuide/Functions.html.

Reply:
There are other diagnostics that mention `void function`:
```
void.swift:1:23: error: non-void function should return a value
func foo() -> Float { return }
                      ^
```

In light of this, I slightly prefer using the current "void function" message.

"Function without return values" is a bit inaccurate because the diagnostic appears for functions that explicitly return the `()` value.

"Function without return type" is also inaccurate, because the return type is inferred to be `Void`.

What do you think?

Reply:
Gotcha, sounds good.